### PR TITLE
GeoPackage loader

### PR DESCRIPTION
**Pull request by kylebarron**
This creates a new `@loaders.gl/sql` module intended for loaders that use SQLite under the hood.

### CODE REVIEW

File: modules/sql/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# @loaders.gl/sql
+
+This module contains helper classes for SQL-based loaders.
```
Comment (ibgreen) [inline modules/sql/README.md:3]: Are we aware of other formats on our potential roadmap that are SQL based? If not we might just call this module `geopackage` for now?

File: modules/sql/src/lib/parse-geopackage.js [hunk 2]
```diff
@@ -0,0 +1,383 @@
+/* global TextDecoder */
```
Comment (ibgreen) [inline modules/sql/src/lib/parse-geopackage.js:1]: Can we fix these? Do we need to install some typings for the SQL module?

**Review by ibgreen — APPROVED** [review]
@belom88 @dryabinin-actionengine - As we discussed, you could potentially add arrow schema generation to this module.

File: modules/sql/README.md (hunk 1 above)
Comment (kylebarron) [reply modules/sql/README.md:3]: No I don't know of other SQL-based formats on the roadmap, but figured naming it as `sql` would have more future compatibility. Happy to rename to `geopackage`

Comment (ibgreen) [reply modules/sql/README.md:3]: Yes let's do that.

**Comment by kylebarron** [comment]
Future work:

- Select a single vector layer/table to load. Right now all vector tables are loaded. This would fit well in the two-stage loader process, where the first stage reads metadata from the file (i.e. the list of available layers) and the second stage loads one or more
- Binary and GeoJSON output. Right now the output is GeoJSON-only, and is contained within an object mapping table names to geometry data. This is the same problem as we discussed previously for MVT, where with GeoPackage especially it's decently likely to only desire a portion of the layers contained in the file.
- More and better tests.

**Pull request merged**